crates/aptos-dkg/src/fiat_shamir.rs

```diff
@@ -29,25 +31,48 @@ pub const PVSS_DOM_SEP: &[u8; 21] = b"APTOS_SCRAPE_PVSS_DST"; // TODO: Name need
 /// ⚠️ This trait is intentionally private: `challenge_scalars`
 /// should **only** be used internally to ensure properly
 /// labelled scalar generation across protocols.
-trait ScalarProtocol {
-    /// **Auxiliary** function to return random scalars
-    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<blstrs::Scalar>;
+pub trait ScalarProtocol<S: FromUniformBytes> {
+    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<S>;
+
+    fn challenge_scalar(&mut self, label: &[u8]) -> S {
+        self.challenge_scalars(label, 1)[0]
+    }
+}
+
+pub trait FromUniformBytes: Copy {
+    const BYTE_SIZE: usize;
+    fn from_uniform_bytes(bytes: &[u8]) -> Self;
+}
+
+/// This reason for this wrapper is to avoid "overlapping implementations" for blstrs::Scalar
+#[derive(Clone, Copy)]
+struct ArkworksScalar<E: Pairing>(E::ScalarField);
+
+impl<E: Pairing> FromUniformBytes for ArkworksScalar<E> {
+    const BYTE_SIZE: usize = (E::ScalarField::MODULUS_BIT_SIZE as usize) / 8;
+
+    fn from_uniform_bytes(bytes: &[u8]) -> Self {
```

Comment:
I think you should assert that `bytes.len() == 2*BYTE_SIZE` here. (For `blstrs`, this is done implicitly in the `try_into` call.)